File: src/lib/software/rustup.py
from src.lib.software import BaseSoftware

class Rustup(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.download_name = 'rustup-init.exe'

    def resolve_download_url(self):
        self.url_resolved.emit('https://static.rust-lang.org/rustup/dist/x86_64-pc-windows-msvc/rustup-init.exe')

File: packages/app/src/services/software/definitions/rustup.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'rustup';
	public name = 'Rustup';
	public category = [SoftwareCategory.Development];
	public downloadName = 'rustup-init.exe';
	public icon = 'rust.png';
	public homepage = 'https://rust-lang.org/learn/get-started';

	public async resolveDownloadUrl() {
		return ok('https://static.rust-lang.org/rustup/dist/x86_64-pc-windows-msvc/rustup-init.exe');
	}
}
